data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -428,8 +428,15 @@ String[] buildCreateTableStatements(@NonNull PersistentEntity entity) {
             traversePersistentProperties(identity, (associations, property) -> {
                 ids.add(PersistentPropertyPath.of(associations, property, ""));
             });
-            if (ids.size() > 1) {
+            int idFieldCount = ids.size();
+            if (idFieldCount > 1) {
```

Comment:
PK generating was missing only for cases when we have one id?

Reply:
Looks like, when id is not composite actually and if does not have `@Generated`.